kratos/python_scripts/project.py

```diff
@@ -0,0 +1,127 @@
+import os
+import sys
+import pickle
+import pathlib
+import importlib
+
+import KratosMultiphysics
+
+class Project:
+    '''Kratos Multiphysics multistage project container.
+
+    This class has two main purposes. First one is to hold the multistage components (output data, active stages and model)
+    Second one is to perform the checkpoint save and load operations.
+    
+    Member variables:
+    __settings -- Kratos parameters object with the multistage simulation settings
+    __output_data -- Dictionary containing the stages data retrieved from GetFinalData
+    __active_stages -- Dictionary containing the active (alive) stage instances
+    __model -- Model instance
+    '''
+
+    def __init__(self, settings : KratosMultiphysics.Parameters) -> None:
+        '''Constructs the multistage project container instance and sets current Kratos version in the settings.'''
+
+        # Declare member variables
+        self.__output_data = {}
+        self.__active_stages = {}
+        self.__settings = settings
+        self.__model = KratosMultiphysics.Model()
+
+        # Add Kratos version and compilation to settings
+        kratos_version = f"{KratosMultiphysics.KratosGlobals.Kernel.Version()}-{KratosMultiphysics.KratosGlobals.Kernel.BuildType()}"
+        self.__settings.AddString("kratos_version", kratos_version)
+
+    def GetModel(self):
+        '''Returns the current multistage simulation model.'''
+
+        return self.__model
+    
+    def GetSettings(self):
+        '''Returns the current multistage simulation settings.'''
+
+        return self.__settings
+    
+    def GetOutputData(self):
+        '''Returns the current multistage simulation output data container.'''
+
+        return self.__output_data
+    
+    def GetActiveStages(self):
+        '''Returns the current multistage simulation active stages dictionary.'''
+
+        return self.__active_stages
+    
+    def AddActiveStage(self, stage_name : str, stage_instance):
+        '''Adds the provided stage instance to the active stages dictionary.'''
+
+        if self.__active_stages.has_key(stage_name):
+            err_msg = f"Stage '{stage_name}' is already active and cannot be added to active stages."
+            raise Exception(err_msg)
+        self.__active_stages[stage_name] = stage_instance
+
+    def RemoveActiveStage(self, stage_name : str):
+        '''Removes an active stage instance from the current stages dictionary.'''
+
+        del self.__active_stages[stage_name]
+
+    def Save(self, save_folder_name : str, checkpoint_file_name : str, output_settings_file_name : str = None):
```

Comment:
```suggestion
    def Save(self, save_folder_name: Path, checkpoint_file_name: Path output_settings_file_name: Optional[Path] = None) -> None:
```

=> requires `from typing import Optional`